### ISSUE TITLE
Duplicate Scanning in HMR Context

### ISSUE THREAD

**Issue by harlan-zw** [windicss/vite-plugin-windicss#166]
When generating CSS via the `generateCSS` function, there is no check for whether or not the scan has already taken place, leading to duplicate scans happening in a HMR context.

For larger projects with many files to scan this can slow down HMR pretty significantly, especially on the webpack plugin.

This affects the vite plugin as well as far as I could tell. From playing around with it, the scan is not required on HMR either. 

I did a dodgy patch to get around the issue here: https://github.com/windicss/windicss-webpack-plugin/commit/d3e65cedcdcac479a2ad437e3047f6d535a5d000

I think if we avoid scanning outside of the init context it should be okay?

**Comment by harlan-zw** [comment]
Hey @antfu, will leave this one for you to take a look at instead of sending a PR. Not sure the best way to solve it

I'd propose something like this

```
   if (!scanned && options.enableScan && options.scanOptions.runOnStartup)
      await scan()
```

**Comment by antfu** [comment]
Should be fixed in 0.16.0-beta.17
